arch/endian_test.go

```diff
@@ -0,0 +1,60 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package arch_test
+
+import (
+	"encoding/binary"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/arch"
+)
+
+type endianTestSuite struct{}
+
+var _ = Suite(&endianTestSuite{})
+
+func (s *endianTestSuite) TestEndian(c *C) {
+	for _, t := range []struct {
+		arch   string
+		endian binary.ByteOrder
+	}{
+		{"ppc", binary.BigEndian},
+		{"ppc64", binary.BigEndian},
+		{"s390x", binary.BigEndian},
+		{"i386", binary.LittleEndian},
+		{"amd64", binary.LittleEndian},
+		{"arm", binary.LittleEndian},
+		{"arm64", binary.LittleEndian},
+		{"riscv64", binary.LittleEndian},
+	} {
+		restore := arch.MockRuntimeGOARCH(t.arch)
+		defer restore()
+
+		c.Check(arch.Endian(), Equals, t.endian)
```

Comment:
Nitpick: this restores all at the end instead of after each test case. It's fine but I think it's less error-prone to not defer in this case (but really there's no practical difference)

```suggestion
		restore := arch.MockRuntimeGOARCH(t.arch)
		c.Check(arch.Endian(), Equals, t.endian)
		restore()
```

Reply:
Thanks, it's a good suggestion and given that Check() is used here this should be safe. I have no strong opinion either way, the more common pattern seems to be the defer but I'm happy either way :)